c++/src/kj/string.h

```diff
@@ -748,7 +748,7 @@ inline ConstString StringPtr::attach() const {
 
 template <typename... Attachments>
 inline ConstString StringPtr::attach(Attachments&&... attachments) const {
-  return ConstString { .content = content.attach(kj::fwd<Attachments>(attachments)...) };
+  return ConstString { content = content.attach(kj::fwd<Attachments>(attachments)...) };
```

Comment:
Does this actually compile? I'd expect this to be a compiler error, since the code now performs an assignment to `this->content`, but `this` is `const`.

I think what you want here is:

```suggestion
  return ConstString(content.attach(kj::fwd<Attachments>(attachments)...));
```

Reply:
Oops yeah good point.
Looks like I was getting the original error without calling this code, so this change wasn't being compiled anyway.
Cleaned it up a bit and moved to a new PR from a new branch: https://github.com/capnproto/capnproto/pull/1898